server/src/main/java/org/spine3/server/event/enrich/EventEnricher.java

```diff
@@ -106,8 +108,8 @@ private EventEnricher(Builder builder) {
      * <p>An event can be enriched if the following conditions are met:
      *
      * <ol>
-     *     <li>There is one or more functions registered for an {@link EnrichmentFunction} where
-     *     the passed class is the {@code source}.
+     *     <li>There is one or more enrichments defined in Protobuf using {@code enrichment_for} and/or {@code by} options.
+     *     <li>There is one or more field enrichment functions registered for the class of the passed event.
```

Comment:
> What happens if there is at least one enrichment_for option and no by references in a type?

An exception is thrown.


>  What happens if the user specifies two or more types in the enrichment_for option? How do we resolve by references then?

Currently, there is no support for this. I can create a task for this and we'll do it later. Or, if this feature is needed in Projects, I'll do it in the next PR.
 We can support such option:

`option (enrichment_for) = "ProjectCreated, ProjectStarted";`

and say that `by` options should be non-qualified.